ui/admin/app/controllers/scopes/scope/roles/role/manage-scopes/index.js

```diff
@@ -31,14 +33,17 @@ export default class ScopesScopeRolesRoleManageScopesIndexController extends Con
   @notifySuccess('resources.role.scope.messages.manage-scopes.success')
   async setGrantScopes(role, grantScopeIDs) {
     await role.setGrantScopes(grantScopeIDs);
+    this.selectedItems = new TrackedArray(role.grant_scope_ids);
     this.router.replaceWith('scopes.scope.roles.role.scopes');
   }
 
   /**
    * Redirect to role scopes as if nothing ever happened.
+   * @param {[string]} grantScopeIDs
    */
   @action
-  cancel() {
+  cancel(grantScopeIDs) {
```

Comment:
Do these cancel actions need to take in any arguments? It looks like you pass in fields from the model which haven't been modified so couldn't you setup the controllers to have access to the models and just use them directly?

Reply:
i was thinking about this. The reason some controller action take in the model as a parameter is because that action is reused in other places. Im leaning towards not using `this.model` in the controller to keep the pattern consistent . Thoughts ?